numba/typed/typeddict.py

```diff
@@ -260,27 +260,59 @@ def box_dicttype(typ, val, c):
 def unbox_dicttype(typ, val, c):
     context = c.context
 
-    miptr = c.pyapi.object_getattr_string(val, '_opaque')
+    # Check that `type(val) is Dict`
```

Comment:
I think this same type-check based fix is also needed in `typed.List` ?

Reply:
right, i'll add that